### PR TITLE
[CORE-2001] Add project defaults messages

### PR DESCRIPTION
**Pull request by robert-uhl**

### CODE REVIEW

File: src/pps/pps.proto [hunk 1]
```diff
@@ -940,6 +940,33 @@ message CreatePipelineTransaction {
   string effective_json = 3; // the effective spec: the result of merging the user JSON into the cluster defaults
 }
 
+message ProjectDefaults {
```
Comment (bbonenfant) [inline src/pps/pps.proto:943]: I'm just curious what the plan is for this message; it's not a component of either the Request or Response messages.

**Review by bbonenfant — APPROVED** [review]

File: src/pps/pps.proto (hunk 1 above)
Comment (robert-uhl) [reply src/pps/pps.proto:943]: It’s not used directly, but is used in JSON-encoded form in the `project_defaults_json` field of the `GetProjectDefaultsResponse` and the `project_defaults_json` field of the `SetProjectDefaultsRequest`.

File: src/pps/pps.proto [hunk 2]
```diff
@@ -940,6 +940,33 @@ message CreatePipelineTransaction {
...
+}
+
+message GetProjectDefaultsRequest {
+  pfs_v2.Project project = 1;
+}
+
+message GetProjectDefaultsResponse {
+  // A JSON-encoded ProjectDefaults message, this is the verbatim input passed
+  // to SetProjectDefaults.
+  string project_defaults_json = 1;
```
Comment (acohen4) [inline src/pps/pps.proto:954]: it makes sense that the pipeline spec is passed in verbatim, but does it also make sense for the Defaults object?

File: src/pps/pps.proto (hunk 1 above)
Comment (acohen4) [inline src/pps/pps.proto:943]: where does the ProjectDefaults Proto get used?

File: src/pps/pps.proto (hunk 2 above)
Comment (robert-uhl) [reply src/pps/pps.proto:954]: I believe so — just like with cluster defaults and the pipeline spec, the user experience is that this is JSON.

**Pull request merged**